python/ray/_raylet.pyx

```diff
@@ -559,6 +559,11 @@ cdef execute_task(
                 task_exception = True
                 raise TaskCancelledError(
                             core_worker.get_current_task_id())
+            if (c_return_ids.size() > 0 and
+                    len(outputs) != int(c_return_ids.size())):
+                raise ValueError(
+                    "Task returned {} objects, but num_returns={}.".format(
```

Comment:
What about adding action items here?

```
Task returned {} objects, but num_returns={}. Consider adding @ray.remote(num_returns={}) to your task/actor decorators.
```

Reply:
This only happens for tasks that already have num returns defined